src/main/java/com/streamr/client/StreamrClient.java

```diff
@@ -122,26 +125,25 @@ public void onMessage(String message) {
                 @Override
                 public void onClose(int code, String reason, boolean remote) {
                     log.info("Connection closed! Code: " + code + ", Reason: " + reason);
-                    state = State.Disconnected;
                     if (remote) {
                         sleepThenReconnect();
                     } else {
+                        state = State.Disconnected;
```

Comment:
Moved state change here to avoid a situation in which someone could invoke `connect` or `publish` (=> `connect`) between `state = State.Disconnected` and `sleepThenReconnect()` thereby causing two simultaneous reconnect attempts.